flink-libraries/flink-table/src/main/scala/org/apache/flink/api/java/table/StreamTableEnvironment.scala

```diff
@@ -164,4 +167,24 @@ class StreamTableEnvironment(
     translate[T](table)(typeInfo)
   }
 
+  /**
+    * Registers a [[TableFunction]] under a unique name in the TableEnvironment's catalog.
+    * Registered functions can be referenced in SQL queries.
+    *
+    * @param name The name under which the function is registered.
+    * @param tf The TableFunction to register
+    */
+  def registerFunction[T](name: String, tf: TableFunction[T]): Unit ={
```

Comment:
The `registerFunction[T: TypeInformation](name: String, tf: TableFunction[T])` only works in Scala. For Java, we need `registerFunction[T](name: String, tf: TableFunction[T])` and help users to extract the type. I don't expect that Java users can see `registerFunction[T: TypeInformation](name: String, tf: TableFunction[T])`.  So I separate this two function into Java/Scala TableEnvironment